doc/debugging.rst

```diff
@@ -14,33 +14,88 @@ This allows breakpoints and the execution flow to be viewed and controlled right
+
+SCREENSHOT: (timeit)
+
+Additionally the debugger allows highlighting of the current frame (debugging step) in the :doc:`editor`.
+
+SCREENSHOT of highlighting step (close up)
+
+Finally, you can write multiline statements while debugging which will allow you execute code while debugging easily.
+
+GIF of multiline debugging
+
+
+
+===========
+Breakpoints
+===========
+
+Spyder's debugger is integrated with a :guilabel:`Breakpoints` pane, listing the file, line, and condition (if any) of every breakpoint defined (:menuselection:`Debug --> List breakpoints`, or :kbd:`Ctrl-Shift-B` by default).
 
 .. image:: images/debugging/breakpoints-standard.png
    :alt: Spyder's Breakpoints panel, with a number of examples showing file, line number and an optional condition
 
-* A :guilabel:`Breakpoints` pane, listing the file, line, and condition (if any) of every breakpoint defined (:menuselection:`&Debug --> List breakpoints`, or :kbd:`Ctrl-Shift-B` by default).
+There are multiple means of setting and clearing normal and conditional breakpoints for any line in a file opened in the :doc:`editor`.
+
+  * By selecting the respective option from the Debug menu.
+  * Through pressing a configurable keyboard shortcut (:kbd:`F12` for normal, or :kbd:`Shift-F12` for conditional breakpoints by default).
+  * By clicking to the left of the line number in an open file in the Editor and :kbd:`Shift`-clicking for conditional breakpoints.
+  * With the ``breakpoint()`` function in your code.
+  * Interactively, using the ``!b`` command in an ``ipdb`` session.
+
+GIF: setting conditional breakpoint shift-clicking
+
+You can access and edit local and global variables at each breakpoint through the :doc:`variableexplorer`.
+
+SCREENSHOT with console and variable explorer showing local and global variables
+
+
+
+=================
+Advanced features
+=================
 
-* Full GUI control over debugger execution from the :guilabel:`Debug` menu, :guilabel:`Debug toolbar` and via configurable keyboard shortcuts, along with the standard ``ipdb`` console commands.
+You can avoid stepping through other Python packages while debugging by enabling the new :guilabel:`Ignore Python libraries while debugging` option in Spyder's preferences under :menuselection:`IPython Console --> Debugger --> Debug`.
+This will skip all the built-in and third-party Python modules.
 
-.. image:: images/debugging/debugging-condbreakpoint.png
-   :alt: Inset of Spyder's Editor, with a breakpoint set and the condition dialog open
+SCREENSHOT with option in preferences
 
-* Highlighting of the current frame (debugging step) in the :doc:`editor`.
+When debugging, if your code has variables with the name of Pdb commands, your debugging workflow might be interrupted.
+To avoid the mix of Pdb commands and Python statements when using the interactive prompt add an exclamation mark in front of the Pdb commands to separate them from Python expressions.
```

Comment:
```suggestion
If your code has variables with the same names as Pdb commands (e.g. ``a``, ``step``), you can still refer to them as normal while debugging.
To call the Pdb command, just add an exclamation point (e.g. ``!a``, ``!step``).
```

* Greatly simplify and clarify this part by cutting out a large amount of repetition and redundancy
* Be more specific and descriptive
* Fix grammar and syntax issues
* Add concrete examples